app/components/FileLoader.js

```diff
@@ -282,8 +286,17 @@ export default class FileLoader extends Component {
       </Table.Row>
     );
 
-    // Generate a row for every file in selected folder
-    const rows = files.map(
+    const fileBeginIndex = (this.props.store.currentPage - 1) * MAX_NUM_OF_FILES_PER_PAGE
+
+    const howManyLeft = (files.length - 1) - fileBeginIndex
+    let lastFileIndex
+    if (howManyLeft > MAX_NUM_OF_FILES_PER_PAGE) {
+      lastFileIndex = fileBeginIndex + MAX_NUM_OF_FILES_PER_PAGE
+    } else {
+      lastFileIndex = fileBeginIndex + howManyLeft
+    }
+
+    const rows = files.slice(fileBeginIndex, lastFileIndex).map(
```

Comment:
What would happen in the case where you're on say, page 5 in a folder, and then you switch to a different folder that has only one page?

Reply:
Need to check. But my guess is, that the fileloader is then completely reloaded, and so these files and pages are calculated again. But yeah need to check, good point